### PR TITLE
Build debug libraries when CMAKE_BUILD_TYPE is set to Debug

### PR DESCRIPTION
**Pull request by dbrnz**
This should address issue #12.

### CODE REVIEW

**Comment by jcfr** [comment]
Thanks for the patch :+1:  I will review and integrate later this week.

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,6 +33,10 @@ if(CMAKE_CROSSCOMPILING)
     cmake_minimum_required(VERSION 3.3) # Version introducing CROSSCOMPILING_EMULATOR
 endif()
 
+if(WIN32 AND "${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
```
Comment (adrianbroher) [inline CMakeLists.txt]: The `CMAKE_DEBUG_POSTFIX` variable only preseeds the `Debug` build type suffix, so there is no need to add that to the condition.

https://cmake.org/cmake/help/v2.8.5/cmake.html#variable:CMAKE_DEBUG_POSTFIX

Comment (dbrnz) [inline CMakeLists.txt]: Thanks @adrianbroher, the redundant check on CMAKE_BUILD_TYPE has been removed.

**Comment by kreuzberger** [comment]
Maybe no good idea to add _d to the debug builds. 
* When using python library in debug config, you have to adapt calling scripts, change linker settings on apps using empedded python and so on. Just take the same name for debug and release, the one who compiles should know what he does
* The line if(MSVC AND "${CMAKE_BUILD_TYPE}" STREQUAL "Debug") will not work for MSVC in multiconfiguration builds. You have to use cmake generater expressions to set the target outputname depending on the build config used during compilation, not on configure.

**Pull request closed without merging**